### PR TITLE
Additional Feature Detectors+

### PR DESCRIPTION
**Pull request by albutko**
Features added:

-  KAZE (OpenCV)
- AKAZE (OpenCV)
- BRISK (OpenCV)
- FAST (OpenCV)
- MSER (OpenCV)
- ORB (OpenCV)
- SuperPoint (Pytorch)

Added `pytorch` as a dependency for the SuperPoint model.

Also created HPatches dset class allowing for comparison on the [hpatches dataset](https://github.com/hpatches/hpatches-dataset). Still a work in progress as there are many more sequences than vggh.

Finally did a few formatting checks on `README.md` and included `python_scores/` and `.pyc` in `.gitignore`.

### CODE REVIEW

**Comment by spongezhang** [comment]
Hi Alex, 

I made a few small changes. Thanks for the work. One comment, can you remove the model file? I would like to keep the rep clean. We can write a download script in __init__() to get it initialized. 

BTW, you should use git add *.py, git add -u to select the file to add. Adding lots of filters in the .gitignore file may not be a good idea. 

Best,
Xu

**Comment by spongezhang** [comment]
Another comment, if I remember correctly, some of the descriptors in OpenCV are binary (BRISK, ORB?). In this case, we can't use Euclidean distance for feature matching. We need to write a new module for Hamming distance.

**Comment by albutko** [comment]
Hi Xu,

I will remove the model file and proceed with only using git add *.py! Were you thinking of adding the Hamming distance to this PR or have this feature in the next?

File: python/bench/ellipse_overlap_H.py [hunk 1]
```diff
@@ -57,6 +57,18 @@ def frame2ellipse(fa):
...
+
+        Args:
+        -   H:
+        -   ell_o:
+        -   method:
+
+        Returns:
+        -   well
+    """
     ell = copy.copy(ell_o)
```
Comment (johnlambert-argo) [inline python/bench/ellipse_overlap_H.py]: Hey @spongezhang, we're trying to make sure we're on the same page for all of these files. Do you think you could add inline comments to these functions and also a description in the function signature with input/output argument types? We can mark the ones we think we most need if that works for you. Thanks so much!

Comment (spongezhang) [reply python/bench/ellipse_overlap_H.py]: This piece of code is a bit tough. It comes from the old vlbenchmark. I may need some time to read the understand it.

**Comment by spongezhang** [comment]
@albutko If you think you fully understood the matching code, I think you can add it right now. Otherwise, it's good to put a todo item here. That's not a easy thing to do. All the distance needs to be changed.

File: python/bench/ellipse_overlap_H.py (hunk 1 above)
Comment (johnwlambert) [reply python/bench/ellipse_overlap_H.py]: @spongezhang Got it -- no worries. Looks like Karel's original code was here: https://github.com/lenck/vlb/blob/master/matlab/%2Butls/ellipse_warp.m. Would be awesome if you could figure out why the warp is carried out this way -- any chance you could ping Karel about that?

Comment (spongezhang) [reply python/bench/ellipse_overlap_H.py]: Hi John,

The code is dated back to some very old codes in vlbenchmark. I believe we can find all the theory in "A Comparison of Affine Region Detectors". 

Some codes in the benchmark is not easy to understand, especially vgg_compute_ellipse_overlap.pyx. Generally, you don't need to care too much about it. 

Best,
Xu

**Pull request merged**